tools/generate_tables_sql_and_testNames.php

```diff
@@ -1,118 +1,234 @@
+                $column_name = str_replace('`', '', $db->escape($bits[1]));
+                $sql_query_columns[] = "`$column_name` $bits[0] default NULL,\n";
+            }
         }
-
+        $output .= "PRIMARY KEY  (`CommentID`)\n) ENGINE=InnoDB"
+            ." DEFAULT CHARSET=utf8;\n";
+        $fp      = fopen($file_name, "w");
+        fwrite($fp, $output);
+        // Check if Table Name exists
+        if (empty($sql_query_table_name)) {
+            $error_message = "Table name does not exist";
+        } else {
+            fwrite(
+                $fp,
+                "REPLACE INTO test_names (Test_name, Full_name, Sub_group) "
+                ."VALUES ('$sql_query_table_name', '$title', 1);\n"
+            );
+            // Writes a record of the changes to a file but is not sourced directly.
+            foreach ($pages as $page_number => $page) {
+                fwrite(
+                    $fp,
+                    "INSERT INTO instrument_subtests "
+                    ."(Test_name, Subtest_name, Description)"
+                    ." VALUES ('$sql_query_table_name', '"
+                    .$sql_query_table_name."_page".($page_number+1)
+                    ."', '$page');\n"
+                );
+            }
+        }
+        // Writes the error message to the sql file.
+        if (!empty($error_message)) {
+            fwrite(
+                $fp,
+                "/* The SQL file is invalid! \n"
+                ."tools/generate_tables_sql_and_testNames.php error_message:\n"
+                ."$error_message\n"
+                ."*/"
+            );
+        }
+        fclose($fp);
     }
-    $output.="PRIMARY KEY  (`CommentID`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8;\n";
-    print "Filename: $filename\n";
-    $fp=fopen($filename, "w");
-    fwrite($fp, $output);
-    fwrite($fp, "REPLACE INTO test_names (Test_name, Full_name, Sub_group) VALUES ('$tablename', '$title', 1);\n");
-    foreach($pages as $pageNo => $page) {
-        fwrite($fp, "INSERT INTO instrument_subtests (Test_name, Subtest_name, Description) VALUES ('$tablename', '" . $tablename  . "_page" . ($pageNo+1) . "', '$page');\n");
-
+    // Execute the SQL tasks if no error_message and CLI flag set.
+    if ($db->isConnected() 
+        && empty($error_message)
+        && $execute_mysql_query
+    ) {
+        if (!$db->tableExists($sql_query_table_name)) {
+            // Table doesn't exist!
+            $sql_query_statement .= implode($sql_query_columns);
+            $sql_query_statement .=
+                "PRIMARY KEY  (`CommentID`)\n) ENGINE=InnoDB "
+                ."DEFAULT CHARSET=utf8;\n";
+            $statement            = $db->prepare($sql_query_statement);
+            $statement->execute();
+        } else {
+            $error_message = "There is already an instrument installed with 
+                this name.";
+        }
+        $statement = $db->prepare(
+            'REPLACE INTO test_names (Test_name, Full_name, Sub_group)'
```

Comment:
another SQL statement duplication. might wanna consider using the same statement appended to the output on line 145.

duplications add the risk of commands diverging in the futur

Reply:
@ridz1208 line 145 is writing to a file that the statement that could potentially be executed or not because the generate_tables_sql_and_testNames.php may be used to execute the sql statement or just build one. Zain told me this was a functionality requirement the generate_tables_sql_and_testNames.php needed when I was fixing that PR from potential sql injection attacks. I don't believe it's a duplicate.

Reply:
line 145:

```
"REPLACE INTO test_names (Test_name, Full_name, Sub_group) "
                ."VALUES ('$sql_query_table_name', '$title', 1);\n"
```

All im saying is that maybe we should not be writing SQL twice in this script. write it once and use it where you need it.

Reply:
I think it's because I found the extra variable for just holding the `REPLACE INTO test_names (Test_name, Full_name, Sub_group)` as bad practice because less variables sometimes is better. 
Specifically line 189 is:
```
$statement = $db->prepare(
            'REPLACE INTO test_names (Test_name, Full_name, Sub_group)'
            .' VALUES (?, ?, 1)'
        );
```
The above is doing a prepare and it only would be able to store `REPLACE INTO test_names (Test_name, Full_name, Sub_group)` as a string and similar to line 145 it would still need the:
`."VALUES ('$sql_query_table_name', '$title', 1);\n"` which seems taste to me. It can be put into a variable if you think it should.